cloud_utils/cache/utils.py

```diff
@@ -20,35 +21,43 @@ class VersionNotFound(Exception):
 
 
 @gamla.curry
-def _write_to_versions_file(identifier: Text, hash_to_load: Text, versions_file):
+def _write_to_cache_file(
+    identifier: str,
+    hash_to_load: str,
+    extra_fields: Dict,
+    cache_file,
+):
     new_versions_dict = gamla.pipe(
-        json.load(versions_file),
+        json.load(cache_file),
         gamla.add_key_value(
             identifier,
             {
-                _HASH_VERSION_KEY: hash_to_load,
+                _RESULT_HASH_KEY: hash_to_load,
                 _LAST_RUN_TIMESTAMP: datetime.datetime.now().isoformat(),
+                **extra_fields,
             },
         ),
         dict.items,
         sorted,
         dict,
     )
-    versions_file.seek(0)
-    json.dump(new_versions_dict, versions_file, indent=2)
-    versions_file.truncate()
+    cache_file.seek(0)
+    json.dump(new_versions_dict, cache_file, indent=2)
+    cache_file.write("\n")
+    cache_file.truncate()
 
 
 @gamla.curry
-def _write_hash_to_versions_file(
-    versions_file_name: Text,
-    identifier: Text,
-    hash_to_load: Text,
+def _write_hash_to_cache_file(
```

Comment:
this function pays 6 signature lines and 5 call lines to save inlining 4 lines.
in other words if you inline 57-60, readability would be the same, 

it would help if _write_to_cache_file has the `hash_to_load` as its last argument, then externally it would look like

`_write_to_cache_file(cache_filename, identifier, extra_fields)`

(can put the `open_file` either inside or inline)

Reply:
Removed function completely